tests/phpunit/tests/option/isEqualDatabaseValue.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Test is_equal_database_value().
+ *
+ * @covers ::is_equal_database_value
+ */
+class Tests_Is_Equal_Database_Value extends WP_UnitTestCase {
+
+	/**
+	 * @ticket 22192
+	 *
+	 * @dataProvider data_is_equal_database_value
+	 *
+	 * @param mixed $old_value The old value to compare.
+	 * @param mixed $new_value The new value to compare.
+	 * @param int   $expected  The expected result.
+	 */
+	public function test_is_equal_database_value( $old_value, $new_value, $expected ) {
+		$this->assertEquals( $expected, is_equal_database_value( $old_value, $new_value ) );
```

Comment:
```suggestion
                 if( $expected ) {
                     $this->assertTrue( is_equal_database_value( $old_value, $new_value ) );
                 } else {
		     $this->assertFalse( is_equal_database_value( $old_value, $new_value ) );
		}
```

Reply:
Definitely agree on replacing `assertEquals()`. Rather than a conditional assertion, we'll see additional benefits if we split the test and data provider into two:
- `test_is_equal_database_value_should_return_true()`
- `test_is_equal_database_value_should_return_false()`.

This means:
- the intent of each test clear.
- that the assertions are clear: `assertTrue()` and `assertFalse()` respectively.
- no repetitive `$expected` parameter needs to be included in each dataset.

Reply:
Let's leave it as it is!